pkg/controller/acm/controller.go

```diff
@@ -0,0 +1,266 @@
+
+	cr, ok := mgd.(*v1alpha1.Certificate)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.RequestCertificateRequest(acm.GenerateCreateCertificateInput(meta.GetExternalName(cr), &cr.Spec.ForProvider)).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+
+	meta.SetExternalName(cr, aws.StringValue(response.RequestCertificateOutput.CertificateArn))
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errPersistExternalName)
+
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+
+	cr, ok := mgd.(*v1alpha1.Certificate)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	// Update Certificate tags
+	if len(cr.Spec.ForProvider.Tags) > 0 {
+
+		desiredTags := make([]awsacm.Tag, len(cr.Spec.ForProvider.Tags))
+		for i, t := range cr.Spec.ForProvider.Tags {
+			desiredTags[i] = awsacm.Tag{Key: aws.String(t.Key), Value: aws.String(t.Value)}
+		}
+
+		currentTags, err := e.client.ListTagsForCertificateRequest(&awsacm.ListTagsForCertificateInput{
+			CertificateArn: aws.String(meta.GetExternalName(cr)),
+		}).Send(ctx)
+
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(acm.IsErrorNotFound, err), errListTagsFailed)
+		}
+
+		if len(desiredTags) != len(currentTags.Tags) {
+			_, err := e.client.RemoveTagsFromCertificateRequest(&awsacm.RemoveTagsFromCertificateInput{
+				CertificateArn: aws.String(meta.GetExternalName(cr)),
+				Tags:           currentTags.Tags,
+			}).Send(ctx)
+			if err != nil {
+				return managed.ExternalUpdate{}, errors.Wrap(err, errRemoveTagsFailed)
+			}
+		}
+		_, err = e.client.AddTagsToCertificateRequest(&awsacm.AddTagsToCertificateInput{
+			CertificateArn: aws.String(meta.GetExternalName(cr)),
+			Tags:           desiredTags,
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errAddTagsFailed)
+		}
+	}
+
+	// Update the Certificate Option
+	if aws.StringValue(cr.Spec.ForProvider.CertificateAuthorityARN) == "" && cr.Spec.ForProvider.CertificateTransparencyLoggingPreference != nil {
+		_, err := e.client.UpdateCertificateOptionsRequest(&awsacm.UpdateCertificateOptionsInput{
+			CertificateArn: aws.String(meta.GetExternalName(cr)),
+			Options:        &awsacm.CertificateOptions{CertificateTransparencyLoggingPreference: *cr.Spec.ForProvider.CertificateTransparencyLoggingPreference},
+		}).Send(ctx)
+
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+		}
+	}
+
+	// Renew the certificate if request for RenewCertificate and Certificate is eligible
+	if aws.BoolValue(cr.Spec.ForProvider.RenewCertificate) {
+
+		if cr.Status.AtProvider.RenewalEligibility == awsacm.RenewalEligibilityEligible {
```

Comment:
Could you give me a brief rundown of when a certificate reports as "renewable". If it doesn't need to be renewed does it still report that it can be?

Reply:
When `ACMPCA` which have principal permission granted issue private certificates and exported can be renewed with this API for [reference](https://docs.aws.amazon.com/sdk-for-go/api/service/acm/#ACM.RenewCertificate).

No, It will not report if the certificate can not be renewed. 
CertificateDetails have a field `RenewalEligibility`  which specifies whether the certificate is eligible to renew the certificate for [reference](https://docs.aws.amazon.com/sdk-for-go/api/service/acm/#CertificateDetail).